### PR TITLE
Feature/plcn 213 data transformation for wall of clients

### PR DESCRIPTION
**Pull request by itm-mne**

### CODE REVIEW

**Review by s-bogelund — APPROVED** [review]

File: src/components/summaries/ClientInfoSummary.tsx [hunk 1]
```diff
@@ -33,16 +27,29 @@ const ClientInfoSummary: FC<ClientSummaryProps> = ({ sx, client }) => {
...
 
 	const urlToDisplay = (): JSX.Element => {
-		if (url && isMedium) {
+		if (clientUrl && isMedium) {
 			return (
 				<>
 					<WebRoundedIcon fontSize="small" />
 					<Typography variant="note" noWrap sx={{ opacity: 0.7, mt: '1px' }}>
-						{titleCase(url ? url : '')}
+						{titleCase(clientUrl ? clientUrl : '')}
```
Comment (itm-mne) [inline src/components/summaries/ClientInfoSummary.tsx:52]: correct these shorthand if's to short-circuit snippets instead

File: src/components/wall-of-clients/ClientListItem.tsx [hunk 2]
```diff
@@ -47,50 +44,100 @@ type PersonTestSummary = {
  * - `AccountManagerInfoSummary`
  * - `DealsStatusSummary`
  */
-const ClientListItem: FC<ClientListItemProps> = ({ clientListItem, children }) => {
-	const { client, suppliers, contactPersons, deal } = clientListItem;
-	const { dealStatus } = deal;
+const ClientListItem: FC<ClientListItemProps> = ({ clientInput }) => {
+	// const { client, suppliers, contactPersons, deal } = clientListItem;
```
Comment (itm-mne) [inline src/components/wall-of-clients/ClientListItem.tsx:48]: delete

File: src/components/wall-of-clients/NestedContactPerson.tsx [hunk 3]
```diff
@@ -1,43 +1,84 @@
...
+import { PersonSummary } from './ClientListItem';
+
+// type NestedContactPersonProps = {
+// 	id: string;
+// 	contactPerson: PersonSummary;
+// 	clientName: string;
+// 	deal: TestDeal;
+// 	onExpand: (id: string) => void;
+// 	isExpanded: boolean;
+// };
```
Comment (itm-mne) [inline src/components/wall-of-clients/NestedContactPerson.tsx:25]: delete

File: src/components/wall-of-clients/NestedContactPerson.tsx [hunk 4]
```diff
@@ -1,43 +1,84 @@
...
-	contactPerson: PersonSummary;
+	isExpanded: boolean;
+	contact: PersonSummary;
 	clientName: string;
-	deal: TestDeal;
+	id: any;
 	onExpand: (id: string) => void;
-	isExpanded: boolean;
 };
 // TODO: Doesn't work yet because parent rerenders when child is expanded
```
Comment (itm-mne) [inline src/components/wall-of-clients/NestedContactPerson.tsx:34]: delete

File: src/components/wall-of-clients/NestedContactPerson.tsx [hunk 5]
```diff
@@ -1,43 +1,84 @@
...
+	// TODO: We need the same data transformation as used in ClientListItem
 
 	return (
 		<Box>
 			<HorizontalDividedContainer
 				isExpandable
 				onExpand={() => onExpand(id)}
 				isExpanded={isExpanded}
-				key={contactPerson.id + Math.random()}
+				key={contact.id + Math.random()}
```
Comment (itm-mne) [inline src/components/wall-of-clients/NestedContactPerson.tsx:81]: remove math.random (unnecessary)

File: src/components/wall-of-clients/NestedContactPerson.tsx [hunk 6]
```diff
@@ -75,7 +116,8 @@ const NestedContactPerson: FC<NestedContactPersonProps> = ({
 								<Stack gap="3px" direction="row" alignItems="center">
 									<NestingIndicator onClick={() => onExpand(id)} height={lineHeight} />
 									<Stack width="100%" ref={nestedContacts}>
-										{testNest()}
+										<Typography>Yoyo, whaddup?</Typography>
+										{/* {testNest()} */}
```
Comment (itm-mne) [inline src/components/wall-of-clients/NestedContactPerson.tsx:120]: Remove

File: src/pages/WallOfClients.tsx [hunk 7]
```diff
@@ -9,16 +9,19 @@ import SecondaryFilterContainer from '../components/common/filters/containers/Se
...
+import {
+	getFilteredClientsQuery,
+	getFilteredClientsQueryVariables,
+} from '../utils/queries/__generated__/wallOfClientsQueries.graphql';
 import { GET_FILTERED_CLIENTS } from '../utils/queries/wallOfClientsQueries';
-/**
- * TODO: Dummy suppliers above^ - remove when real data is available
- */
+
+// TODO: Dummy suppliers above^ - remove when real data is available
```
Comment (itm-mne) [inline src/pages/WallOfClients.tsx:24]: Rename, to specify that its the dummy data and not the line above the comment

File: src/utils/extractMostRelevantDeal.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
+import { FRAGMENT_DEALFragment } from './queries/__generated__/wallOfClientsQueries.graphql';
+
+/**
+ * Takes an array of Deal-objects and checks whether any deal has a specific status in the following order: Active, Dialog, Inactive. Returns the most relevant Deal from the highest priority, non-empty array.
+ */
+export const extractMostRelevantDeal = (dealsArray: FRAGMENT_DEALFragment[]) => {
```
Comment (itm-mne) [inline src/utils/extractMostRelevantDeal.ts:6]: Add return-type

File: src/utils/extractMostRelevantDeal.ts [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+	}
+
+	const inactiveDeals: FRAGMENT_DEALFragment[] = dealsArray.filter(
+		deal => deal.dealStatus === 'InActive'
+	);
+	if (inactiveDeals) {
+		return extractDeals(inactiveDeals);
+	}
+
+	return undefined;
```
Comment (itm-mne) [inline src/utils/extractMostRelevantDeal.ts:28]: Is it necessary to return undefined? Maybe it is better to just return an empty array. This means that we dont need to handle the undefined return type.

**Review by LarsITMinds — APPROVED** [review]

**Pull request merged**